### ISSUE TITLE
moor_ffi => Plugin request for plugin already on the classpath must not include a version

### ISSUE THREAD

**Issue by SebastianKlaiber** [simolus3/moor#256]
I get this error message when I try to add moor_ffi 0.2.0
Is there any other way for unit testing?

FAILURE: Build failed with an exception.

* Where:
Build file '/flutter/.pub-cache/hosted/pub.dartlang.org/moor_ffi-0.2.0/android/build.gradle' line: 13

* What went wrong:
Error resolving plugin [id: 'de.undercouch.download', version: '4.0.0']
> Plugin request for plugin already on the classpath must not include a version

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1sGradle task assembleStagingDebug failed with exit code 1

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, v1.11.0, on Mac OS X 10.14.6 18G95, locale de-DE)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 11.2.1)
[✓] Android Studio (version 3.5)
[✓] IntelliJ IDEA Community Edition (version 2019.2)
[✓] VS Code (version 1.40.1)
[✓] Connected device (1 available)

• No issues found!

**Comment by simolus3** [comment]
If you're only using moor_ffi for testing, I'm wondering if maybe making it a `dev_dependency` could help. But I'm not sure about that.

Either way, it looks like this is caused in combination with another Flutter plugin. Can you share the dependencies you use?

**Comment by SebastianKlaiber** [comment]
I have also tried it as a dev_dependency but with the same result.

Here are my dependencies:
  share: ^0.6.3+1
  cupertino_icons: ^0.1.2
  firebase_core: ^0.4.0+9
  firebase_messaging: ^5.1.6
  image_picker: ^0.6.1+4
  flutter_bloc: ^2.0.1
  bloc_test: ^2.1.0
  equatable: ^0.6.1
  built_value: ^6.7.0
  built_collection: ^4.2.2
  dio: ^2.1.13
  faker: ^1.1.1
  intl: ^0.16.0
  shared_preferences: ^0.5.3+4
  uuid: ^2.0.2
  path_provider: ^1.2.0
  image: ^2.1.4
  flushbar: ^1.9.1
  get_it: ^3.0.1
  corsac_jwt: ^0.1.2
  logger: ^0.7.0+2
  logger_flutter: ^0.7.1
  cached_network_image: ^2.0.0-rc.1
  sentry: any
  device_info: ^0.4.0+2
  package_info: ^0.4.0+6
  connectivity: ^0.4.5+2
  flutter_svg: ^0.14.0
  uni_links: ^0.2.0
  rflutter_alert: ^1.0.3
  flutter_masked_text: ^0.8.0
  flutter_datetime_picker: ^1.2.6
  floating_action_row: ^0.1.0
  auto_size_text: ^2.1.0
  camera: ^0.5.6+3
  sliding_panel: ^0.7.0
  flutter_exif_rotation: ^0.2.6
  division: ^0.8.3

**Comment by SebastianKlaiber** [comment]
I have also tried with a new project but get the same error. Could it be due to Flutter 1.11.0?

**Comment by simolus3** [comment]
I don't think so. It's probably because another package also uses the download gradle plugin, so moor_ffi can't declare an explicit version in its `build.gradle`. I'll see what can be done about that, maybe we can just remove the plugin and instead upload the sqlite source code to pub.

**Comment by simolus3** [comment]
I've removed the gradle plugin on the `develop` branch, so it should work in the next release. You can preview the changes by putting this in your pubspec:
```yaml
dependency_overrides:
  moor_ffi:
    git:
      url: https://github.com/simolus3/moor.git
      ref: develop
      path: moor_ffi
```

I also realized that the error when using it as a `dev_dependency` means that Flutter tries to include the native libraries whenever it's building your app. If you're not using moor_ffi outside of tests, that's a waste of app size for no reason :( I'll check if there's something that can be done about this or create an issue on Flutter.

**Comment by SebastianKlaiber** [comment]
Tried it yesterday so far it worked. Will give you feeback if I notice anything else.

**Comment by SebastianKlaiber** [comment]
Tried it yesterday so far it worked. Will give you feeback if I notice anything else.

**Comment by simolus3** [comment]
Should be fixed in version 0.3.0 of `moor_ffi`, so you can remove the `dependency_overrides` section for moor.
